cli/azd/internal/appdetect/spring_boot_property.go

```diff
@@ -134,5 +135,8 @@ func getEnvironmentVariablePlaceholderHandledValue(rawValue string) string {
 		placeholder := match[0]
 		result = strings.Replace(result, placeholder, value, -1)
 	}
+	if result == "#{null}" {
+		result = ""
+	}
```

Comment:
Originally, I add this to support this scenario:
> spring.data.mongodb.uri=${AZURE_COSMOS_CONNECTION_STRING:#{null}}

`#{null}` is SPEL.

This idea comes from here:
https://github.com/Azure-Samples/todo-java-mongo-aca/blob/d8fff510c9185c6ec4754a6671c2776a51b5668d/src/api/src/main/resources/application.properties#L10
> ![image](https://github.com/user-attachments/assets/9ac28598-7075-4832-863f-39861ee0421d)

But when I debug it, I found that SPEL is not supported in `DataObjectPropertyBinder`.
> ![image](https://github.com/user-attachments/assets/4af7fccc-417b-4f09-b239-f3d8e425b160)

I'll remove this.